src/testTree.ts

```diff
@@ -144,6 +144,8 @@ export class TestTree extends DisposableBase {
         vsChildrenById.set(id, vsChild);
         vsChildren.add(vsChild);
       }
+      if (uChild.kind === 'case')
+        vsChild.tags = uChild.tags.map(tag => new this._vscode.TestTag(tag));
```

Comment:
Let's compare them to the existing ones first in order to not trigger update every time.